mmcv/transform/processing.py

```diff
@@ -0,0 +1,382 @@
+import random
+import warnings
+from typing import Dict, List, Optional, Sequence, Tuple, Union
+
+import numpy as np
+
+import mmcv
+from mmcv.utils import build_from_cfg
+from .base import BaseTransform
+from .builder import TRANSFORMS
+from .wrappers import Compose
+
+Number = Union[int, float]
+
+
+@TRANSFORMS.register_module()
+class CenterCrop(BaseTransform):
+    """Center crop the image. If the crop area exceeds the original image and
+    ``pad_mode`` is not None, then pad the original image first.
+
+    Args:
+        crop_size (Union[int, Tuple[int, int]]):  Expected size after cropping
+            with the format of (h, w).
+        pad_val (Union[Number, Dict[str, Number]], optional): A dict for
+            padding value. Defaults to dict(img=0, seg=255).
+        pad_mode (str, optional): Padding mode. Defaults to None.
+    """
+
+    def __init__(self,
+                 crop_size: Union[int, Tuple[int, int]],
+                 pad_val: Union[Number, Dict[str,
+                                             Number]] = dict(img=0, seg=255),
+                 pad_mode: Optional[str] = None):
+        super().__init__()
+        assert isinstance(crop_size, int) or (isinstance(crop_size, tuple)
+                                              and len(crop_size) == 2)
+        if isinstance(crop_size, int):
+            crop_size = (crop_size, crop_size)
+        assert crop_size[0] > 0 and crop_size[1] > 0
+        self.crop_size = crop_size
+        self.pad_val = pad_val
+        self.pad_mode = pad_mode
+
+    def transform(self, results: dict) -> dict:
```

Comment:
Can split to `def _crop_img(self, results)` `def _crop_seg-map` for inheritance in downstream repo